### ISSUE TITLE
Don't let rule PP fail

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#19679]
When pushing pages/sites, we include rules as dependencies. This can cause conflicts and cause bundles to fail.  We should be more resilient to such failures whenever there is a conflict of rules.

This might involve using upserts or skipping failed inserts.

- A known case is where a rule's actionlet or conditionlet is dependent on an osgi plugin that does not exist on the receiver.  In such as case, we should just skip the osgi based actionlet and/or conditionlet

We only see this error when a rule is included by dependency, from the host

**Comment by nollymar** [comment]
PR Enterprise: https://github.com/dotCMS/enterprise/pull/766
PR Core: https://github.com/dotCMS/core/pull/19699

**Comment by fabrizzio-dotCMS** [comment]
Verified on branch `release-21.20`
I was able to verify that push publishing pages with rules depending on OSGI plugins (actionlet/conditionlet) that don't exist on the receiver are copied just fine. I tried setting up a rule with various conditions where some conditions were OSGI dependent while others were not. And only the offending ones are ignored while the others show just as expected. 

![Screen Shot 2021-01-14 at 17 10 17](https://user-images.githubusercontent.com/37148755/104661630-8f08de80-568e-11eb-8c1d-9881567b5cbe.png)


When pushing a rule that has an offending dependency (like in this case) the Publishing queue shows a new status like "Success ith Warnings" 
![Screen Shot 2021-01-14 at 17 07 47](https://user-images.githubusercontent.com/37148755/104661599-7b5d7800-568e-11eb-808d-148ef37f4ecb.png)

When visualizing a rule that depends on a non-existing plugin  on the receiver I get the following warning

`WARN  exception.InternalServerException - {"error":"dotcms.api.error.internal_server_error: Could not create ReST Rule from Server, Rule id '88b7d6de-cf26-4644-b086-39421ef6c9ba'"}
com.dotcms.rest.exception.InternalServerException: HTTP 400 Bad Request
	at com.dotcms.rest.api.v1.sites.ruleengine.rules.RuleTransform.appToRest(RuleTransform.java:117) ~[dotcms_21.02_ccff39c.jar:?]
	at com.dotcms.rest.api.v1.sites.ruleengine.rules.RuleResource.getRulesInternal(RuleResource.java:222) ~[dotcms_21.02_ccff39c.jar:?]
	at com.dotcms.rest.api.v1.sites.ruleengine.rules.RuleResource.list(RuleResource.java:84) ~[dotcms_21.02_ccff39c.jar:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_162]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_162]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_162]`


But it looks fine on the UI.

Upon missing plugins installation the rules are rendered properly. No need to push the pages from the receiver.
